### DISCUSSION TITLE
Improve osu!lazer music playing ability

### DISCUSSION THREAD

**Discussion by developomp** [ppy/osu#16929] [Ideas] [answered]
The osu beatmap list is a free, curated list of quality music many of which I can listen to for an entire day. I used to search and download each music which was such a pain in the ass, then I used youtube playlist which allowed me to skip the downloading part but it had its own limitations. This is why now I always have osu!lazer opened in the background. I don't have to search for the right music, however, there still are two issues with this method:

First, the music doen't start from the beginning. I can always drag the progress bar in the "Now playing" menu (F6) to the beginning but I have to do this each time the music ends.

Second, the lack of looping feature. If I want to listen to one music continuously I have to select a music then open the "Now playing" menu. If I want to listen to random music I then have to close the menu and go to the start screen. Both method does not start the music from the beggining as mentioned in point NO.1.

Here's my solution for this problem:

- Activate "Music Mode" when the "Now Playing" menu is opened. This will play the music from start to end without skipping any part.
- Add a loop button with 3 modes:
  - Single - Loops one music from start to end
  - Sequential - Loop over all music in the selected collection (collection is already implemented)
  - Random - Plays a random music based on the selected random algorithm (This is also implemented in the "Song Select" option)

**Comment by bdach** [comment, marked as answer]
osu! is not a music player, period. This is out of scope, sorry.

See:
* https://github.com/ppy/osu/discussions/13244
* https://github.com/ppy/osu/issues/7306
* https://github.com/ppy/osu/issues/6546
* https://github.com/ppy/osu/issues/4263
* https://github.com/ppy/osu/issues/3409
